sp_api/base/client.py

```diff
@@ -44,8 +44,8 @@ def __init__(
             version=None,
             credential_providers=None,
     ):
-        if os.environ.get('SP_API_DEFAULT_MARKETPLACE', None):
-            marketplace = Marketplaces[os.environ.get('SP_API_DEFAULT_MARKETPLACE')]
+        if not marketplace and (default_marketplace:=os.environ.get('SP_API_DEFAULT_MARKETPLACE')):
```

Comment:
marketplaces is set to a default value in the constructor, this will only be true if marketplaces is set to None explicitly. That's not compatible with previous versions, can't be merged. 

Still has to be changed for sure though, the argument marketplace should always override the default, not sure where this was introduced.